### PR TITLE
migrate tests in detekt-api to junit

### PR DESCRIPTION
**Pull request by marschwar**
This is part on #4450.

Unfortunately it was too late when I realized it will collide with #4518. 

@BraisGabin Feel free to mark this as blocked if #4518 should be merged first.

### CODE REVIEW

**Comment by BraisGabin** [comment]
Don't worry, it would be really easy to fix.

File: detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/SingleAssignSpec.kt [hunk 1]
```diff
@@ -2,34 +2,36 @@ package io.gitlab.arturbosch.detekt.api
...
-internal class SingleAssignSpec : Spek({
-    describe("value is unset") {
-        var unassigned: Int by SingleAssign()
-        it("should fail when value is retrieved") {
-            assertThatIllegalStateException().isThrownBy {
-                @Suppress("UNUSED_EXPRESSION")
-                unassigned
-            }
-        }
+@TestInstance(PER_METHOD)
```
Comment (BraisGabin) [inline detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/SingleAssignSpec.kt:9]: This is the default, right? We don't need it.

**Review by BraisGabin — APPROVED** [review]

File: detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/SingleAssignSpec.kt (hunk 1 above)
Comment (marschwar) [reply detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/SingleAssignSpec.kt:9]: No, we changed the setting for All Tests to per-class and have to change it here.

Comment (BraisGabin) [reply detekt-api/src/test/kotlin/io/gitlab/arturbosch/detekt/api/SingleAssignSpec.kt:9]: Oh! You are right! my bad.

**Pull request merged**